## 50_joc-toggle-case.diff by Peter S Galbraith <psg@debian.org>

--- a/elisp/emacs-goodies-el/joc-toggle-case.el
+++ b/elisp/emacs-goodies-el/joc-toggle-case.el
@@ -28,7 +28,9 @@
 ;; Boston, MA 02111-1307, USA.
 ;;; **************************************************************************
 
-;;; Description:
+;;; Commentary:
+;; 
+;; Description:
 ;;
 ;;  This packages provides a sophisticated (over-engineered?) set of
 ;;  functions to toggle the case of the character under point, with
@@ -38,16 +40,16 @@
 ;;  character, allowing successive invocations to progress down the
 ;;  line.
 
-;;; Installation:
+;; Installation:
 ;;
 ;;  Put this file on your Emacs-Lisp load path and add the following to your
 ;;  ~/.emacs startup file
 ;;
-;;     (require 'toggle-case)
+;;     (require 'joc-toggle-case)
 ;;
 ;;  See below for key-binding suggestions.
 
-;;; Usage:
+;; Usage:
 ;;
 ;;  M-x `joc-toggle-case'
 ;;     Toggles the case of the character under point.  If called with
@@ -81,7 +83,7 @@
 ;;  M-x `joc-toggle-case-by-word-backwards'
 ;;     Toggles the case of all characters in the current region.
 
-;;; Customization:
+;; Customization:
 ;;
 ;;  M-x `joc-toggle-case-customize' to customize all package options.
 ;;
@@ -95,7 +97,7 @@
 ;;        is reversed, the semantics of this are reveresed as well
 ;;        (i.e. does it stop at the beginning of the line).
 
-;;; Keybinding examples:
+;; Keybinding examples:
 ;;
 ;;  This is what I have -- use it or not as you like.
 ;;
@@ -119,7 +121,7 @@
 ;;  Any comments, suggestions, bug reports or upgrade requests are welcome.
 ;;  Please send them to Joe Casadonte (emacs@northbound-train.com).
 ;;
-;;  This version of toggle-case was developed and tested with NTEmacs
+;;  This version of joc-toggle-case was developed and tested with NTEmacs
 ;;  2.7 under Windows NT 4.0 SP6 and Emacs 20.7.1 under Linux (RH7).
 ;;  Please, let me know if it works with other OS and versions of Emacs.
 
@@ -128,6 +130,14 @@
 ;;; **************************************************************************
 ;;; **************************************************************************
 ;;; **************************************************************************
+
+;;; History:
+;; 
+;;  2003-11-23 Peter S Galbraith <psg@debian.org>
+;;   This version, distributed in the Debian package `emacs-goodies-el',
+;;   was renamed from toggle-case.el to joc-toggle-case.el.  The prefix
+;;   was also added in the file where appropriate.
+
 ;;; Code:
 
 ;;; **************************************************************************
@@ -139,7 +149,7 @@
 
 ;; ---------------------------------------------------------------------------
 (defun joc-toggle-case-customize ()
-  "Customization of the group joc-toggle-case."
+  "Customization of the group `joc-toggle-case'."
   (interactive)
   (customize-group "joc-toggle-case"))
 
@@ -150,7 +160,7 @@
 stop at the end of the line, set to `nil' it will not (it
 will continue on to the next line).  If direction of toggle
 is reversed, the semantics of this are reveresed as well
-(i.e. does it stop at the beginning of the line)."
+\(i.e. does it stop at the beginning of the line)."
   :group 'joc-toggle-case
   :type 'boolean)
 
@@ -163,26 +173,27 @@
 
 ;; ---------------------------------------------------------------------------
 (defun joc-toggle-case-version-number ()
-  "Returns joc-toggle-case version number."
+  "Return `joc-toggle-case' version number."
   (string-match "[0123456789.]+" joc-toggle-case-version)
   (match-string 0 joc-toggle-case-version))
 
 ;; ---------------------------------------------------------------------------
 (defun joc-toggle-case-display-version ()
-  "Displays joc-toggle-case version."
+  "Displays `joc-toggle-case' version."
   (interactive)
   (message "joc-toggle-case version <%s>." (joc-toggle-case-version-number)))
 
 ;;; **************************************************************************
 ;;; ***** interactive functions
 ;;; **************************************************************************
+;;;###autoload
 (defun joc-toggle-case (prefix)
-  "Toggles the case of the character under point.  If called with
-a prefix argument, it toggles that many characters (see
-joc-toggle-case-stop-at-eol).  If the prefix is negative, the
-case of the character before point is toggled, and if called
-with a prefix argument, N characters before point will have
-their case toggled (see also joc-toggle-case-backwards)."
+  "Toggle the case of the character under point.
+If called with a PREFIX argument, it toggles that many
+characters (see joc-toggle-case-stop-at-eol).  If the prefix is
+negative, the case of the character before point is toggled, and
+if called with a prefix argument, N characters before point will
+have their case toggled (see also joc-toggle-case-backwards)."
 
   (interactive "*p")
 
@@ -207,14 +218,16 @@
 		  (setq lcv count)))))
 
 ;; ---------------------------------------------------------------------------
+;;;###autoload
 (defun joc-toggle-case-backwards (prefix)
-  "Convenience function to toggle case of character preceeding
-point.  This is the same as calling joc-toggle-case with a
-negative prefix (and is in fact implemented that way)."
+  "Convenience function to toggle case of character preceeding point.
+This is the same as calling joc-toggle-case with a negative
+prefix (and is in fact implemented that way)."
   (interactive "*p")
   (joc-toggle-case (- prefix)))
 
 ;; ---------------------------------------------------------------------------
+;;;###autoload
 (defun joc-toggle-case-by-word (prefix)
   "Similar to joc-toggle-case except that the count (supplied by
 the prefix argument) is of the number of words, not letters, to
@@ -238,14 +251,16 @@
 	(joc-toggle-case (- end start))))
 
 ;; ---------------------------------------------------------------------------
+;;;###autoload
 (defun joc-toggle-case-by-word-backwards (prefix)
-  "Convenience function to toggle case by word, backwards.  This
-is the same as calling joc-toggle-case-by-word with a
+  "Convenience function to toggle case by word, backwards.
+This is the same as calling joc-toggle-case-by-word with a
 negative prefix (and is in fact implemented that way)."
   (interactive "*p")
   (joc-toggle-case-by-word (- prefix)))
 
 ;; ---------------------------------------------------------------------------
+;;;###autoload
 (defun joc-toggle-case-by-region (start end)
   "Toggles the case of all characters in the current region."
   (interactive "*r")
@@ -310,8 +325,6 @@
 ;;; **************************************************************************
 ;;; ***** we're done
 ;;; **************************************************************************
-(provide 'toggle-case)
+(provide 'joc-toggle-case)
 
-;; toggle-case.el ends here!
-;;; **************************************************************************
-;;;; *****  EOF  *****  EOF  *****  EOF  *****  EOF  *****  EOF  *************
+;;; joc-toggle-case.el ends here
